stacker/tests/test_plan2.py

```diff
@@ -0,0 +1,241 @@
+            'namespace-vpc.1', 'namespace-db.1'])
+
+    def test_execute_plan_exception(self):
+        vpc = Stack(
+            definition=generate_definition('vpc', 1),
+            context=self.context)
+        bastion = Stack(
+            definition=generate_definition('bastion', 1, requires=[vpc.fqn]),
+            context=self.context)
+
+        calls = []
+
+        def fn(stack, status=None):
+            calls.append(stack.fqn)
+            if stack.fqn == vpc_step.name:
+                raise ValueError('Boom')
+            return COMPLETE
+
+        vpc_step = Step(vpc, fn)
+        bastion_step = Step(bastion, fn)
+        plan = build_plan(description="Test", steps=[vpc_step, bastion_step])
+
+        with self.assertRaises(ValueError):
+            plan.execute()
+
+        self.assertEquals(calls, ['namespace-vpc.1'])
+
+    def test_execute_plan_skipped(self):
+        vpc = Stack(
+            definition=generate_definition('vpc', 1),
+            context=self.context)
+        bastion = Stack(
+            definition=generate_definition('bastion', 1, requires=[vpc.fqn]),
+            context=self.context)
+
+        calls = []
+
+        def fn(stack, status=None):
+            calls.append(stack.fqn)
+            if stack.fqn == vpc_step.name:
+                return SKIPPED
+            return COMPLETE
+
+        vpc_step = Step(vpc, fn)
+        bastion_step = Step(bastion, fn)
+
+        plan = build_plan(description="Test", steps=[vpc_step, bastion_step])
+        plan.execute()
+
+        self.assertEquals(calls, ['namespace-vpc.1', 'namespace-bastion.1'])
+
+    def test_build_plan_missing_dependency(self):
+        bastion = Stack(
+            definition=generate_definition(
+                'bastion', 1, requires=['namespace-vpc.1']),
+            context=self.context)
+
+        with self.assertRaises(GraphError) as expected:
+            build_plan(description="Test", steps=[Step(bastion, None)])
+        message = ("Error detected when adding 'namespace-vpc.1' "
+                   "as a dependency of 'namespace-bastion.1': node "
+                   "namespace-vpc.1 does not exist")
+        self.assertEqual(expected.exception.message, message)
+
+    def test_build_plan_cyclic_dependencies(self):
+        vpc = Stack(
+            definition=generate_definition(
+                'vpc', 1),
+            context=self.context)
+        db = Stack(
+            definition=generate_definition(
+                'db', 1, requires=['namespace-app.1']),
+            context=self.context)
+        app = Stack(
+            definition=generate_definition(
+                'app', 1, requires=['namespace-db.1']),
+            context=self.context)
+
+        with self.assertRaises(GraphError) as expected:
+            build_plan(
+                description="Test",
+                steps=[Step(vpc, None), Step(db, None), Step(app, None)])
+        message = ("Error detected when adding 'namespace-db.1' "
```

Comment:
You know, you probably wouldn't need this message test if we had different exceptions ;)  Just saying.